src/dotprod/src/dotprod.c

```diff
@@ -60,6 +60,9 @@ void DOTPROD(_run)(TC *         _h,
 //  _x      :   input array [size: 1 x _n]
 //  _n      :   input lengths
 //  _y      :   output dot product
+#if defined(LIQUID_FMV) && (__GNUC__ >= 6)
+__attribute__((target_clones("sse", "mmx", "avx", "default")))
```

Comment:
maybe this is why it did not preform so well, in target clones it matters in what order you list the options since it will use the first valid one (sse probably in this case)

maybe try this
```c
__attribute__((target_clones("avx", "sse", "mmx", "default")))
```

Reply:
> maybe try this
> 
> __attribute__((target_clones("avx", "sse", "mmx", "default")))

thanks for your suggestion, just gave it  a run on my i7 machine. Not much of a difference in performance results. Looks like then compilers are still not that smart when it comes to auto-vectorization.